lib/repo_tools.pm

```diff
@@ -558,4 +559,20 @@ sub verify_software {
     return '';
 }
 
+=head2 ensure_ca_certificates_suse_installed
+    ensure_ca_certificates_suse_installed();
+
+This functions checks if ca-certificates-suse is installed and if it is not it adds the repository and installs it.
+
+=cut
+
+sub ensure_ca_certificates_suse_installed {
```

Comment:
I suggest to actually rename the method so that it does make sense to be called in all cases regardless if "SUSE certificates" are required. How about `ensure_repo_certificates_installed`? For openSUSE the function would then immediately return because the function does not need to do anything as all necessary "repo certificates" are already installed.

Reply:
openSUSE is shipped with suse internal CA?

Reply:
We use this function within openqa which need to download stuff from ibs and for this it is convenient to have these certs installed also on tumbleweed.

Reply:
No, sorry, I had not make myself clear. You already have the code `return unless (is_sle || is_tumbleweed || is_leap);`. I was suggesting to rename the function and change that one return line so that it's harmless to call it on openSUSE, e.g. like this:

```perl
sub ensure_repo_certificates_installed {
    return unless is_sle;
    ...
```

Reply:
but actually that is the point. I need the certificates also on openSUSE and other distros.

Reply:
> No, sorry, I had not make myself clear. You already have the code `return unless (is_sle || is_tumbleweed || is_leap);`. I was suggesting to rename the function and change that one return line so that it's harmless to call it on openSUSE, e.g. like this:
> 
> ```perl
> sub ensure_repo_certificates_installed {
>     return unless is_sle;
>     ...
> ```

@okurz , @cfconrad  what about moving to installing certificates from OBS ? I know about [this](https://build.opensuse.org/package/show/openSUSE:Tools:GoCD/ca-certificates-suse) , but one might say that it is not official enough ? I actually never understand why we having something like this only in internal repo ...

Reply:
I like it!

Reply:
well, would be nice if the certificates could be just public and provided upstream, yes.